modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessExecuteAction.java

```diff
@@ -82,40 +119,187 @@ public Response newResponse() {
         return new Response();
     }
 
-    public static class Request extends ActionRequest implements ToXContent {
+    public static class Request extends SingleShardRequest<Request> implements ToXContent {
 
         private static final ParseField SCRIPT_FIELD = new ParseField("script");
-        private static final ParseField CONTEXT_FIELD = new ParseField("context");
+        private static final ParseField EXECUTE_CONTEXT_FIELD = new ParseField("execute_context");
```

Comment:
I think think this name doesn't convey the meaning. Execute context could easily be a normal script context. But this is really using a script context, but providing some arguments to it. I think perhaps just context is better, but maybe the context name could be separated from a separate argument like `context_setup` that is parsed based on which context is chosen?